src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,24 +88,52 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def azure_keyvault_backend(
+        url: str,
+        client: str,
+        secret: str,
+        tenant: str,
+        secret_field: str,
+        secret_version: str,
+) -> str:
```

Comment:
MyPy says that `.value` can also be `None`. So the correct return type is
```suggestion
) -> str | None:
```

If you want/need to keep it `str`, you'll need to make sure you don't ever return `None`.